File: UnityFramework/Assets/Frame/Scripts/GameStart/GameStart.cs
namespace WKC
{
    public class GameStart : Singleton<GameStart>
    {
        private void Awake()
        {
            ConfigManager.Instance.Init(FrameInit);
        }

        private void FrameInit(params object[] args)
        {
            AtlasManager.Instance.Init();
            UIManager.Instance.Init();
            ObjectPoolManager.Instance.Init();
            StartGame();
        }

        private void StartGame()
        {
            DataStorage.Instance.LoadJsonsData();
            LoadSceneManager.Instance.LoadSceneAsync("Level1", (args) =>
            {
                TimeManager.Instance.UpdatePerSecondEventHandler += SaveDataToLocal;
                UIManager.Instance.ShowPanel(PanelName.MainMenuPanel);
            });
        }


        private void SaveDataToLocal()
        {
            DataStorage.Instance.userdata.gold++;
            DataStorage.Instance.SaveJsonDataToLocal();
        }
    }
}

File: UnityFramework/Assets/Lua/GameStart.lua
require("Other/UIManager")
require("Other/GlobalVariable")
require("Other/DoTweenManager")

UIStart = {}

UIStart.Start = function()

	print("UIStart")

	--CS.LoadSceneManager.Instance:LoadSceneAsync("Level1",function ( ... )
		-- body
		--CS.TimeManager.Instance.UpdatePerSecondEventHandler += SaveDataToLocal;
		UIManager:ShowPanel("MainMenuPanel");
	--end);
--[[
	UIManager:ShowPanel("BGPanel");

	local loginPanel = UIManager:ShowPanel("LoginPanel");
	require("Controller/LoginController");
	LoginController:Start(loginPanel);
]]--
end
